src/mats_l1_processing/L1_calibration_functions.py

```diff
@@ -944,86 +937,75 @@ def calculate_time_per_row(header):
+    expected pixel value and the measured one in the artifact. Several azimuth
+    angles intervals are defined and the corresponding mask is applied to the
+    image.
+
     Arguments:
         ccditem : Panda series
             Panda series containing the ccditem
         image (optional) : np.array
-            If this is given then it will be used instead of the image in the CCDitem
-        
+            If this is given then it will be used instead of the image in the
+            CCDitem
+
     Returns:
         corrected_image : np.array(float.64)
-            numpy array representing the corrected image  
+            numpy array representing the corrected image
         error_flag : np.array(float.64)
             numpy array representing the corrected pixels
     """
 
-
-
     if image is None:
-        image = ccditem["IMAGE"]   
-               
+        image = ccditem["IMAGE"]
+
     artifact_masks = ccditem['CCDunit'].get_artifact_mask()
 
-    if len(artifact_masks)==1 and np.sum(np.abs(artifact_masks.iloc[0]['bias_mask'])==0.0): # if an empty mask is applied
-        warnings.warn("Empty mask applied (no correction applied)")
-        corrected_im = image
-        error_flag_mask = np.zeros(corrected_im.shape, dtype=np.uint16)
-        error_flag_no_correction = np.ones(corrected_im.shape, dtype=np.uint16)        
-        error_flag = combine_flags([error_flag_mask,error_flag_no_correction],[1,1])
-
-        return(corrected_im,error_flag)
-
-    MASK_AZIMUTH = artifact_masks['azimuth'] # list of all the azimuth values in the dataframe
-    m = len(artifact_masks)
-
-    try : 
-        azimuth = nadir_az(ccditem) # nadir azimuth angle of the ccditem
-    except: # if no azimuth angle can be calculated, no correction is applied
-        warnings.warn("Unable to compute the nadir solar azimuth angle (no correction applied)")
-        corrected_im = image
-        error_flag_mask = np.zeros(corrected_im.shape, dtype=np.uint16)
-        error_flag_no_correction = np.ones(corrected_im.shape, dtype=np.uint16)        
-        error_flag = combine_flags([error_flag_mask,error_flag_no_correction],[1,1])
-        return(corrected_im,error_flag)
-
-    # finding the mask which corresponding azimuth angle interval is the closest to the image's azimuth angle
-    distance = 360.0
-    best_ind = 0
-    for j in range(m):
-        if abs(MASK_AZIMUTH[j]-azimuth) < distance:
-            best_ind = j
-            distance = abs(MASK_AZIMUTH[j]-azimuth)
+    error_flag_mask = np.zeros_like(image, dtype=np.uint16)
+    error_flag_no_correction = np.ones_like(image, dtype=np.uint16)
+    error_flag = combine_flags([error_flag_mask, error_flag_no_correction], [1, 1])
+
+    # if an empty mask is applied
+    if (
+        len(artifact_masks) == 1
+        and np.sum(np.abs(artifact_masks.iloc[0]['bias_mask']) == 0.0)
+    ):
+        warnings.warning("Empty mask applied (no correction applied)")
+        return corrected_im, error_flag
```

Comment:
it should be  : return image, error_flag

Reply:
Fixed